src/Operators/remapping.jl

```diff
@@ -138,110 +179,82 @@ function overlap_weights(
-    source::S,
-) where {
-    T <: SpectralElementSpace2D{<:Topology2D, Quadratures.GL{1}},
-    S <: SpectralElementSpace2D{<:Topology2D, Quadratures.GL{1}},
-}
-    # Calculate element overlap pattern in x-dimension
-    # X_ov[i,j] = overlap length along x-dimension between target elem i and source elem j
-    X_ov = fv_x_overlap(target, source)
-
-    # Calculate element overlap pattern in y-dimension
-    Y_ov = fv_y_overlap(target, source)
-
-    return kron(Y_ov, X_ov)
-end
-
-function fv_x_overlap(
-    target::T,
-    source::S,
-) where {
-    T <: Union{SpectralElementSpace1D, SpectralElementSpace2D},
-    S <: Union{SpectralElementSpace1D, SpectralElementSpace2D},
-}
-    target_topo = Spaces.topology(target)
-    source_topo = Spaces.topology(source)
-    ntarget, nsource = nxelems(target_topo), nxelems(source_topo)
-    v1, v2 = 1, 2
-
-    W_ov = spzeros(ntarget, nsource)
-    for i in 1:ntarget
-        vertices_i = Topologies.vertex_coordinates(target_topo, i)
-        min_i, max_i = xcomponent(vertices_i[v1]), xcomponent(vertices_i[v2])
-        for j in 1:nsource
-            vertices_j = Topologies.vertex_coordinates(source_topo, j)
-            min_j, max_j =
-                xcomponent(vertices_j[v1]), xcomponent(vertices_j[v2])
-            min_ov, max_ov = max(min_i, min_j), min(max_i, max_j)
-            overlap_length = max_ov > min_ov ? max_ov - min_ov : continue
-            W_ov[i, j] = overlap_length
-        end
-    end
-    return W_ov
-end
-
-function fv_y_overlap(
+function y_overlap(
     target::T,
     source::S,
 ) where {
     T <: Union{SpectralElementSpace1D, SpectralElementSpace2D},
     S <: Union{SpectralElementSpace1D, SpectralElementSpace2D},
 }
+    FT = Spaces.undertype(source)
     target_topo = Spaces.topology(target)
     source_topo = Spaces.topology(source)
-    ntarget, nsource = nyelems(target_topo), nyelems(source_topo)
+    nelems_t, nelems_s = nyelems(target_topo), nyelems(source_topo)
     nx_target, nx_source = nxelems(target_topo), nxelems(source_topo)
     elem_idx = (i, n) -> 1 + (i - 1) * n
+    QS_t = Spaces.quadrature_style(target)
+    QS_s = Spaces.quadrature_style(source)
+    Nq_t = Quadratures.degrees_of_freedom(QS_t)
+    Nq_s = Quadratures.degrees_of_freedom(QS_s)
     v1, v2 = 1, 4
 
-    W_ov = spzeros(ntarget, nsource)
-    for i in 1:ntarget
+    J_ov = spzeros(nelems_t * Nq_t, nelems_s * Nq_s)
+    for i in 1:nelems_t
         vertices_i =
             Topologies.vertex_coordinates(target_topo, elem_idx(i, nx_target))
         min_i, max_i = ycomponent(vertices_i[v1]), ycomponent(vertices_i[v2])
-        for j in 1:nsource
+        for j in 1:nelems_s
             vertices_j = Topologies.vertex_coordinates(
                 source_topo,
                 elem_idx(j, nx_source),
             )
             min_j, max_j =
                 ycomponent(vertices_j[v1]), ycomponent(vertices_j[v2])
             min_ov, max_ov = max(min_i, min_j), min(max_i, max_j)
```

Comment:
Everything from this line to the end of the function is shared with `x_overlap` - can box in another method or maybe two(?), one for the calculation of the interpolation matrices with basis function values at the overlap nodes, and one to fill in `W_ov` via the integration rule.